interfaces/repo.go

```diff
@@ -753,5 +753,20 @@ func (r *Repository) AutoConnectCandidates(plugSnapName, plugName string) []*Slo
 // isAutoConnectCandidate returns true if the plug is a candidate to
 // automatically connect to the given slot.
 func isAutoConnectCandidate(plug *Plug, slot *Slot) bool {
-	return slot.Snap.Type == snap.TypeOS && slot.Interface == plug.Interface
+	// content sharing auto connect candidates
```

Comment:
Not sure if this would help but perhaps the essence of the auto-connection feature could be clearer to follow if this was a switch statement:

```
switch slot.Interface {
    case "content":
      ...
   default:
        return slot.Snap.Type == snap.TypeOS && slot.Interface == plug.Interface
}